app/models/article.rb

```diff
@@ -629,6 +634,16 @@ def canonical_url_must_not_have_spaces
     errors.add(:canonical_url, "must not have spaces")
   end
 
+  def user_mentions_in_markdown
```

Comment:
This method does something _extremely` similar to what we do in the `Comment` model, but I intentionally chose to not extract it because of the rule of threes. However, if we find ourselves doing this again (a third time), I think we _should_ extract it into something nicer. At the moment, the way I have it also allows us to configure a different `MAX_USER_MENTIONS` down the road, if we want to.

Reply:
How many concepts support mentions? Is it just articles and comments?

Reply:
It's only comments right now. But soon, it _will also include_ articles. But that will be in some upcoming, separate PRs (see this [very involved](https://github.com/forem/rfcs/pull/22) RFC, lol 😅)

Reply:
Oh, right, we talked about that. 🙃 Sorry, I'm still trying to wrap my mind around this all (and I'm admittedly bad at reading comprehension). :joy: 

To be transparent, the purpose of that question was to understand if the rule of threes applies in this particular case. If we only have 2-ish and we want them to work the same, it may make more sense to go ahead and link that functionality together.

I'm not sure how to really convey what I mean, but maybe like this: instead of duplicating a single chunk of code 3 times, if we duplicate 3 different chunks of code for the same reasons, does that count?

(also, to be clear, this isn't to say I think it should be done in this PR specifically — it probably warrants its own)

Reply:
Haha, no worries! I think that's a fair question; I do think it might make a lot of sense to extract it, but I think we might want to wait until the @-mention functionality for articles is complete first before doing so. I wrote a reply [here](https://github.com/forem/forem/pull/13259#discussion_r607401247) to your other comment that might kind of serve as a reply to this follow up question, too 😃

It's almost like we're having a real-time conversation lol 😂